cli/azd/pkg/tools/github/github.go

```diff
@@ -449,20 +473,25 @@ func downloadGh(
 	extractImplementation extractGitHubCliFromFileImplementation,
 	path string) error {
 
+	binaryName := func(platform string) string {
+		return fmt.Sprintf("gh_%s_%s", ghVersion, platform)
+	}
+
+	systemArch := runtime.GOARCH
 	// arm and x86 not supported (similar to bicep)
 	var releaseName string
 	switch runtime.GOOS {
 	case "windows":
-		releaseName = "gh_2.22.1_windows_amd64.zip"
+		releaseName = binaryName(fmt.Sprintf("windows_%s.zip", systemArch))
 	case "darwin":
-		releaseName = "gh_2.22.1_macOS_amd64.tar.gz"
+		releaseName = binaryName(fmt.Sprintf("macOS_%s.tar.gz", systemArch))
```

Comment:
Might need some updates here based on 'macOS archives attached to our releases are no longer .tar.gz, but .zip instead. This is because .tar.gz archives cannot be notarized.' - https://github.com/cli/cli/releases/tag/v2.28.0

Reply:
good catch.
I have updated it to zip.  We use `zip` for Windows, so the support is already there for zip and tar.gz